.github/actions/install_node_dependencies/action.yml

```diff
@@ -0,0 +1,22 @@
+name: Setup Node.js and Install Dependencies
+description: Sets up a specific Node.js version, caches Node modules, and installs Node dependencies.
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/setup-node@v4.0.0
+      with:
+        node-version: 16.13.2
```

Comment:
[Nit] To make this reusable may be we can have node-version as input ?

Reply:
Done.